index.bs

```diff
@@ -4358,6 +4424,237 @@ Of the extended attributes defined in this specification, only the [{{Exposed}}]
+
+The [=partial module declaration/module declaration members|members=] of a
+[=partial module declaration=] must not be [=list/is empty|empty=].
+
+[=Operations=] must not be [=overloaded=] across [=module declarations=] and [=partial module
+declarations=].
+
+[=Module members=] and [=interface members|members=] of [=interfaces=] whose
+[=interface/enclosing module=] is not null must not be annotated with the [{{Exposed}}] or
+[{{SecureContext}}] [=extended attributes=].
+
+Note: The restrictions on the [{{Exposed}}] and [{{SecureContext}}] [=extended attributes=] in
+this section could be relaxed if there is a sufficient use case;
+they are restricted here for simplicity of specification and implementation, and a
+lack of clear use cases. Please <a
+href="https://github.com/heycam/webidl/issues/new?title=Fine%20grained%20exposure%20in%20modules">
+file an issue</a> if you see a need for this feature.
+
+<div algorithm>
+  The <dfn for="module declaration,partial module declaration"
+  lt="module declaration members">members</dfn> of a [=module declaration=] or [=partial module
+  declaration=] are defined as follows:
+
+    1.  Let |members| be « ».
+    1.  For each string |s| matching the
+        <emu-nt><a href="#prod-ModuleMember">ModuleMember</a></emu-nt> nonterminal symbol between
+        the braces of the [=definition=]:
+        1.  If |s| matches <emu-t>interface</emu-t>
+            <emu-nt><a href="#prod-InterfaceRest">InterfaceRest</a></emu-nt>:
+            1.  [=list/Append=] the [=interface=] defined by |s| to |members|.
+        1.  If |s| matches <emu-t>partial</emu-t> <emu-t>interface</emu-t>
+            <emu-nt><a href="#prod-PartialInterfaceRest">PartialInterfaceRest</a></emu-nt>:
+            1.  [=list/Append=] the [=partial interface=] defined by |s| to |members|.
+        1.  If |s| matches
+            <emu-nt><a href="#prod-RegularOperation">RegularOperation</a></emu-nt>:
+            1.  Let |operation| be the [=operation=] defined by |s|.
+            1.  If |members| [=list/contains=] a [=/set=] of [=operations=] whose [=identifier=] is
+                |operation|'s [=identifier=], then:
+                1.  Let |set| be that [=/set=].
+            1.  Otherwise:
+                1.  Let |set| be a new [=/set=].
+                1.  [=list/Append=] |set| to |members|.
+            1.  [=set/Append=] |operation| to |set|.
+        1.  If |s| matches <emu-t>readonly</emu-t>
+            <emu-nt><a href="#prod-AttributeRest">AttributeRest</a></emu-nt>:
+            1.  [=list/Append=] the [=attribute=] defined by |s| to |members|.
+    1.  Return |members|.
+</div>
+
+<div data-fill-with="grammar-Partial"></div>
+
+<div data-fill-with="grammar-PartialDefinition"></div>
+
+<pre class="grammar" id="prod-Module">
+    Module :
+        "module" identifier "{" ModuleMembers "}" ";"
+</pre>
+
+<pre class="grammar" id="prod-PartialModule">
+    PartialModule :
+        "module" identifier "{" ModuleMembers "}" ";"
```

Comment:
Why do we have two exactly same grammars `Module` and `PartialModule`?

Reply:
To distinguish the two cases in the definition of "module declaration" and "partial module declaration". There may be a better way to do this.

Reply:
I think the following two sentences are ambiguous become the grammars are same:

* A module declaration is a definition matching the Module nonterminal symbol.
* A partial module declaration is a definition matching the PartialModule nonterminal symbol.